### DISCUSSION TITLE
Best model/config for running in low freq (2.5ghz) multiple core xeon cpu (cpu only, no gpu).

### DISCUSSION THREAD

**Discussion by S1gil0** [oobabooga/text-generation-webui#1881] [General]
I have available a bunch of cores to experiment in a 2.5 xeon linux server. Llama and vicuña (7b and 13b) are using only 4 cores, is there another model that I can use that works better on cpu only mode?

**Comment by m-spangenberg** [comment]
Have you tried passing `--cpu --threads #your-core-count-here` when starting the server?

**Comment by S1gil0** [comment]
i was missing the --threads ty so much. 
eight 2.5 xeon vcores are running vicuña 7b at 4 tokens per second.
i will try more cores and provide feedback.

**Reply by SLiX-KAOS** [reply]
did this work and where did you make this change to add more cores/threads ? in the webui.py file?

**Reply by m-spangenberg** [reply]
Take a look at the [README](https://github.com/oobabooga/text-generation-webui#llamacpp), those are all flags you can use. You would start the server with, example: `python3 server.py --cpu --threads 8 --model-menu`, to start oobabooga ui in CPU inference mode with 8 cores. You will need a [GGML model](https://huggingface.co/TheBloke/guanaco-7B-GGML) to make use of `llama.cpp` and I believe `llama.cpp` uses 4 cores by default.
